lib/ansible/modules/crypto/openssl_pkcs12.py

```diff
@@ -0,0 +1,333 @@
+        self.friendly_name = module.params['friendly_name']
+        self.passphrase = module.params['passphrase']
+
+    def generate(self, module):
+        ''' Generate PKCS#12 file archive. '''
+
+        file_args = module.load_file_common_arguments(module.params)
+        this_mode = module.params['mode']
+        if not this_mode:
+            this_mode = int('0400', 8)
+
+        if not self.check(module, perms_required=False) or module.params['force']:
+            self.pkcs12 = crypto.PKCS12()
+
+            try:
+                self.remove()
+            except PkcsError as exc:
+                module.fail_json(msg=to_native(exc))
+
+            if self.ca_certificates:
+                ca_certs = [crypto_utils.load_certificate(ca_cert) for ca_cert
+                            in self.ca_certificates]
+                self.pkcs12.set_ca_certificates(ca_certs)
+
+            if self.cert_path:
+                self.pkcs12.set_certificate(crypto_utils.load_certificate(
+                                            self.cert_path))
+
+            if self.friendly_name:
+                self.pkcs12.set_friendlyname(self.friendly_name)
+
+            if self.privatekey_path:
+                self.pkcs12.set_privatekey(crypto_utils.load_privatekey(
+                                           self.privatekey_path,
+                                           self.privatekey_passphrase)
+                                           )
+
+            try:
+                with open(self.path, 'wb', this_mode) as archive:
+                    archive.write(
+                        self.pkcs12.export(
+                            self.passphrase,
+                            self.iter_size,
+                            self.maciter_size
+                        )
+                    )
+                self.changed = True
+            except (IOError, OSError) as exc:
+                self.remove()
+                raise PkcsError(exc)
+
+        if module.set_fs_attributes_if_different(file_args, False):
+            module.set_mode_if_different(self.path, this_mode, False)
+            self.changed = True
+
+    def check(self, module, perms_required=True):
+        ''' Ensure the resource is in its desired state. '''
+
+        state_and_perms = super(Pkcs, self).check(module, perms_required)
+
+        def _check_pkey_passphrase():
+            if self.privatekey_passphrase:
+                try:
+                    crypto_utils.load_privatekey(self.path,
+                                                 self.privatekey_passphrase)
+                    return True
+                except crypto.Error:
+                    return False
+            return True
+
+        if not state_and_perms:
+            return state_and_perms
+
+        return _check_pkey_passphrase
+
+    def dump(self):
+        ''' Serialize the object into a dictionary. '''
+
+        result = {
+            'changed': self.changed,
+            'filename': self.path,
```

Comment:
Could you add one row here so we know which privatekey has been used for that (if we register the module output)

Like this: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/crypto/openssl_publickey.py#L235